SegmentStats yaml test fails in mixed-cluster BWC

Looks like it started failing after the [backport version change](https://github.com/elastic/elasticsearch/commit/b8f8ed7f89405b630d73505972d79e06befc8a59#diff-1280f38c9141fb09ef09489d91856342):

![image](https://user-images.githubusercontent.com/1224228/54782951-83654180-4bf6-11e9-87d8-7157311f10d5.png)

Seems to be failing pretty consistently so I'm going to mute the test.

```
 ./gradlew :qa:mixed-cluster:v7.1.0#mixedClusterTestRunner -Dtests.seed=D6DF7185080F7FEE -Dtests.class=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT -Dtests.method="test {p0=indices.stats/30_segments/Segment Stats}" -Dtests.security.manager=true -Dtests.locale=el-CY -Dtests.timezone=America/Halifax -Dcompiler.java=11 -Druntime.java=8
```
```
[2019-03-21T15:32:20,409][INFO ][o.e.b.MixedClusterClientYamlTestSuiteIT] [test] [p0=indices.stats/30_segments/Segment Stats] before test
14:34:45   1> [2019-03-21T15:32:20,625][INFO ][o.e.b.MixedClusterClientYamlTestSuiteIT] [test] Stash dump on test failure [{
14:34:45   1>   "stash" : {
14:34:45   1>     "num_segments_after_flush" : 1,
14:34:45   1>     "body" : {
14:34:45   1>       "_shards" : {
14:34:45   1>         "total" : 1,
14:34:45   1>         "successful" : 0,
14:34:45   1>         "failed" : 0
14:34:45   1>       },
14:34:45   1>       "_all" : {
14:34:45   1>         "primaries" : { },
14:34:45   1>         "total" : { }
14:34:45   1>       },
14:34:45   1>       "indices" : { }
14:34:45   1>     },
14:34:45   1>     "num_segments" : 0
14:34:45   1>   }
14:34:45   1> }]
14:34:45   1> [2019-03-21T15:32:20,681][INFO ][o.e.b.MixedClusterClientYamlTestSuiteIT] [test] [p0=indices.stats/30_segments/Segment Stats] after test
14:34:45 FAILURE 0.28s | MixedClusterClientYamlTestSuiteIT.test {p0=indices.stats/30_segments/Segment Stats} <<< FAILURES!
14:34:45    > Throwable #1: java.lang.AssertionError: Failure at [indices.stats/30_segments:53]: field [indices.test.primaries.segments.count] is null
14:34:45    > 	at __randomizedtesting.SeedInfo.seed([D6DF7185080F7FEE:5E8B4E5FA6F31216]:0)
14:34:45    > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:393)
14:34:45    > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:370)
14:34:45    > 	at java.lang.Thread.run(Thread.java:748)
14:34:45    > Caused by: java.lang.AssertionError: field [indices.test.primaries.segments.count] is null
14:34:45    > 	at org.elasticsearch.test.rest.yaml.section.MatchAssertion.doAssert(MatchAssertion.java:79)
14:34:45    > 	at org.elasticsearch.test.rest.yaml.section.Assertion.execute(Assertion.java:76)
14:34:45    > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:386)
14:34:45    > 	... 37 more
```

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/indices.stats/30_segments.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/indices.stats/30_segments.yml
@@ -1,9 +1,5 @@
 ---
 setup:
-  - skip:
-      version: "all"
-      reason: "AwaitsFix: https://github.com/elastic/elasticsearch/issues/40331"
-
   - do:
       indices.create:
           index: test
@@ -47,6 +43,7 @@ setup:
   - do:
       indices.close:
         index: test
+        wait_for_active_shards: all
 
   - do:
       indices.stats:

